src/client.ts

```diff
@@ -0,0 +1,184 @@
+import {BigInteger, default as bigInt} from 'big-integer';
+import {Config} from './config';
+import {hash, randomSalt} from './crypto';
+import {Group} from './groups';
+import {Identity} from './identity';
+import {KeyPair} from './keypair';
+import {multiplier} from './multiplier';
+import {zeroLeftPad} from './zero-left-pad';
+
+interface Verifier {
+    username: string;
+    salt: string;
+    verifier: string;
+}
+
+export class Client {
+    private readonly identity: Identity;
+    private readonly config: Config;
+    private readonly salt: Buffer;
+    constructor(identity: Identity, config: Config) {
```

Comment:
I think we shouldn't accept from the client these internal classes. 

```suggestion
    constructor(username: string, password: string, options: OptionsInterface) {
```

We should expose to the client, the simplest API possible, how we do it internally it's up to us. To the so, we must not expose to him classes like Identity or Config, he should pass literal values.

```js
constructor(username: string, password: string, options: OptionsInterface) {
        this.identity = new Identity(username, password);
        this.config = new Config(options);
        this.salt = randomSalt();
}
```